sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java

```diff
@@ -55,7 +55,7 @@ static HttpPipeline buildPipeline(AzureNamedKeyCredential azureNamedKeyCredentia
                                       HttpLogOptions logOptions, ClientOptions clientOptions, HttpClient httpClient,
                                       List<HttpPipelinePolicy> perCallAdditionalPolicies,
                                       List<HttpPipelinePolicy> perRetryAdditionalPolicies, Configuration configuration,
-                                      ClientLogger logger) {
+                                      ClientLogger logger, boolean enableTenantDiscovery) {
```

Comment:
What's the scenario where users would like to disable tenant discovery?

Reply:
Because this adds an extra unauthorized request for the initial service request, it could cause unwanted failures for customers. For example, a 403 in their telemetry.